packages/magic/src/index.ts

```diff
@@ -0,0 +1,206 @@
+            )
+          }
+        }
+
+        const handleLogin = async () => {
+          loggedIn = await loginModal({
+            walletName: walletName,
+            brandingHTMLString: brandingHTML,
+            emailLoginFunction: loginWithEmail
+          })
+        }
+
+        if (!loggedIn) await handleLogin()
+
+        let magicProvider = magicInstance.rpcProvider
+        let provider: EIP1193Provider
+
+        function patchProvider(): EIP1193Provider {
+          provider = createEIP1193Provider(magicProvider.enable(), {
+            eth_requestAccounts: async () => {
+              try {
+                if (!loggedIn) await handleLogin()
+                const accounts = await magicProvider.request({
+                  method: 'eth_accounts'
+                })
+                return accounts
+              } catch (error) {
+                console.error('error in request accounts', error)
+                const { code } = error as { code: number }
+                if (code === -32603) {
+                  throw new ProviderRpcError({
+                    code: ProviderRpcErrorCode.ACCOUNT_ACCESS_REJECTED,
+                    message: 'Account access rejected'
+                  })
+                }
+                return []
+              }
+            },
+            eth_selectAccounts: null,
+            eth_getBalance: async () => {
+              const address = (
+                await magicProvider.request({
+                  method: 'eth_accounts'
+                })
+              )[0]
+              const balance = await magicProvider.request({
+                method: 'eth_getBalance',
+                params: [address, 'latest']
+              })
+              return balance
+                ? BigNumber.from(balance).mul('1000000000000000000').toString()
+                : '0'
+            },
+            wallet_switchEthereumChain: async ({ params }) => {
+              const chain = chains.find(({ id }) => id === params[0].chainId)
+              if (!chain) throw new Error('Chain must be set before switching')
+              currentChain = chain.id
+              // re-instantiate instance with new network
+              customNodeOptions = {
+                chainId: parseInt(chains[0].id, 10),
+                rpcUrl: chains[0].rpcUrl
+              }
+
+              magicInstance = new Magic(apiKey, {
+                network: customNodeOptions
+              })
+
+              magicProvider = magicInstance.rpcProvider
+
+              emitter.emit('chainChanged', chain.id)
+
+              patchProvider()
+
+              return null
+            },
+            eth_accounts: async () => {
+              const accounts = (
+                await magicProvider.request({
+                  method: 'eth_accounts'
+                })
+              )
+              return Array.isArray(accounts) && accounts.length
+                ? [accounts[0]]
+                : []
+            },
```

Comment:
```suggestion
```

I believe that we can just remove this and not patch it as it seems like the method is to spec? Is there a situation where it does not return an array?

Reply:
Yep you are correct, removed!